_core/page/head.php

```diff
@@ -43,111 +41,77 @@ function generate() {
-                    <meta http-equiv='expires' content='0' />
-                    <meta http-equiv='expires' content='Tue, 01 Jan 1980 1:00:00 GMT' />
-                    <meta http-equiv='pragma' content='no-cache' />
-                    <link rel='shortcut icon' href='" . CSS_PATH . "imgs/favicon.ico'>
-                    <title>" . $this->info['page']['title'] . "</title>";
-
-		$cs = array_keys($cssArray);
-		foreach ($cssArray as $key => $value) {
-			$type = (NSFW) ? (($key == $cs[0]) ? "stylesheet" : "alternate stylesheet") : ($key == $cs[1]) ? "stylesheet" : "alternate stylesheet" ;
-			$dat .= "<link class='togglesheet' rel='$type' type='text/css' href='" . CSS_PATH . $value . "' title='" . $key . "' />";
-		}
+        foreach($this->info['js']['raw'] as $script) //Adding js code in <script tags> to the head
+            $dat .= "<script type='text/javascript'>{$raw}</script>";
 
-		$dat .= $this->headerJS(true);
-        $dat .= "<script src='" . JS_PATH . "/jquery.min.js' type='text/javascript'></script>
-                <script src='" . JS_PATH . "/main.js' type='text/javascript'></script>
-				<script src='" . JS_PATH . "/jquery-ui-1.10.4.min.js' type='text/javascript'></script>
-				<script src='" . JS_PATH . "/admin.js' type='text/javascript'></script></head>";
-
-        if (!$noHead) {
-            $dat .= '<div class="beforePostform" />' . $titlebar . '
-                    <span class="boardList desktop">' . ((file_exists(BOARDLIST)) ? file_get_contents(BOARDLIST) : '') . '</span></div>
-                    <div class="linkBar">[<a href="' . HOME . '" target="_top">' . S_HOME . '</a>][<a href="' . PHP_ASELF_ABS . '">' . S_ADMIN . '</a>]</div>
-                    <div class="boardBanner">' . $bannerImg . $boardTitle . '</div>';            
-            
-            $dat .= "<div class='panelOps' style='text-align:left;' />[<a href=\"" . PHP_SELF2 . "\">" . S_RETURNS . "</a>][<a class='cmd' title='Update the index' href='javascript:;' data-cmd='update-index'>Update</a>]";
-
-            if (valid('moderator')) {
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=rebuild' title='Rebuild all pages' >Rebuild all</a>]";
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=all' >Deletion panel</a>]";
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=reports' >" . $getReport->countBoard() . "</a>]";
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=blist' >Bans</a>]";
-            }
-            if (valid('admin')) {
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=staff' >Users</a>]";
-                $dat .= "[<a href='" . PHP_ASELF_ABS . "?mode=news' >Edit News/Boardlist</a>]";
-            }
-            $dat .= "[<a href='" . PHP_ASELF . "?mode=logout'>" . S_LOGOUT . "</a>]</div>";
+        foreach($this->info['js']['script'] as $script) //Adding scripts to the head
+            $dat .= "<script src='" . JS_PATH . "/{$script}' type='text/javascript'></script>";
```

Comment:
````PHP
$dat .= "<script src='" . JS_PATH . "/{$script}' type='text/javascript'></script>";
````

{ } in a double-quoted (") line forces evaluation of the variable if it may conflict. Are { } required? 

For instance, `{$script}omatic` would be required.